source/code-snippets/find.go

```diff
@@ -0,0 +1,46 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+// Replace the uri string with your MongoDB deployment's connection string.
+const uri = "mongodb+srv://<username>:<password>@<cluster-address>/test?w=majority"
+
+func main() {
+	client, err := mongo.Connect(context.TODO(), options.Client().ApplyURI(uri))
+
+	if err != nil {
+		panic(err)
+	}
+	defer func() {
+		if err = client.Disconnect(context.TODO()); err != nil {
+			panic(err)
+		}
+	}()
+
+	// begin find
+	myCollection := client.Database("sample_training").Collection("zips")
+	filter := bson.D{{"pop", bson.D{{"$lte", 500}}}}
+
+	cursor, err := myCollection.Find(context.TODO(), filter)
+	// end find
+
+	if err != nil {
+		panic(err)
+	}
+
+	var results []bson.M
```

Comment:
We normally avoid `bson.M` for regular use-cases since it's unordered and use `bson.D` instead. 
```suggestion
	var results []bson.D
```